### PR TITLE
Add map() documentation sample

### PR DESCRIPTION
**Pull request by borsini**
Small code sample for the collections.map() function

### CODE REVIEW

**Comment by ilya-g** [comment]
It's worth to provide different samples for `Map.map` and `CharSequence/String.map`, because their usages are slightly different.

File: libraries/stdlib/samples/test/samples/collections/collections.kt [hunk 1]
```diff
@@ -393,6 +393,12 @@ class Collections {
             assertPrints(chars.joinToString(limit = 5, truncated = "...!") { it.toUpperCase().toString() }, "A, B, C, D, E, ...!")
         }
 
+        @Sample
+        fun map() {
+            val numbers = listOf(75, 111, 116, 108, 105, 110)
+            assertPrints(numbers.map { it.toChar() }, "[K, o, t, l, i, n]")
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/collections/collections.kt]: I'd consider using some more simple operation, for example squaring numbers or taking `length` property of string elements.

Comment (borsini) [reply libraries/stdlib/samples/test/samples/collections/collections.kt]: Sure I will do that 👍

**Comment by borsini** [comment]
As suggested, I added samples for Map and CharSequence. I also simplified the first example.

**Comment by borsini** [comment]
@ilya-g do the changes look good to you ?

File: libraries/stdlib/samples/test/samples/collections/maps.kt [hunk 2]
```diff
@@ -303,6 +302,12 @@ class Maps {
             assertPrints(props.getProperty("z", "fail"), "fail")
         }
 
+        @Sample
+        fun mapMap() {
+            val peopleToAge = mapOf("alice" to 20, "bob" to 21)
+            assertPrints(peopleToAge.map { it.value }, "[20, 21]")
```
Comment (ilya-g) [inline libraries/stdlib/samples/test/samples/collections/maps.kt]: I propose to add another example with entry argument destructuring:
`peopleToAge.map { (name, age) -> "$name is $age years old" }`

File: libraries/stdlib/common/src/generated/_Maps.kt [hunk 3]
```diff
@@ -58,6 +58,8 @@ public inline fun <K, V, R, C : MutableCollection<in R>> Map<out K, V>.flatMapTo
 /**
  * Returns a list containing the results of applying the given [transform] function
  * to each entry in the original map.
+ * 
+ * @sample samples.collections.Maps.Transformations.mapMap
```
Comment (ilya-g) [inline libraries/stdlib/common/src/generated/_Maps.kt]: to many `map`s in the sample qualified name :)
`mapToList` would be better.

File: libraries/stdlib/samples/test/samples/collections/maps.kt (hunk 2 above)
Comment (borsini) [reply libraries/stdlib/samples/test/samples/collections/maps.kt]: Indeed, I placed it first as it is the obvious way to use it.

File: libraries/stdlib/common/src/generated/_Maps.kt (hunk 3 above)
Comment (borsini) [reply libraries/stdlib/common/src/generated/_Maps.kt]: Done

**Comment by ilya-g** [comment]
Thanks, I've merged it.

**Pull request merged**